### PR TITLE
coordinator,gc: delete stale cluster service safepoint

### LINKED ISSUE #4610: TiCDC may leave a stale service GC safepoint after all changefeeds are deleted and the last CDC node is removed
### What did you do?

Based on source inspection, there appears to be a lifecycle gap in TiCDC's cluster-level service GC safepoint handling.

Scenario:

1. Create one or more changefeeds.
2. Stop a changefeed so that it still blocks GC via its checkpoint.
3. Delete all changefeeds in the TiCDC cluster.
4. Remove or shut down the last TiCDC node shortly after that.

Relevant code paths:

- `ChangeFeedInfo.NeedBlockGC()` still returns `true` for `StateStopped`.
- `ChangefeedDB.CalculateGlobalGCSafepoint()` only ignores a changefeed after it is removed from the in-memory changefeed set.
- When there is no remaining changefeed, `coordinator.updateGlobalGcSafepoint()` does not delete the TiCDC service safepoint. Instead, it sets the safepoint to `currentTSO - 1`.
- I could not find a normal coordinator shutdown path that explicitly removes the long-lived TiCDC service safepoint.

This means the cluster-level TiCDC service safepoint can remain in PD even after all changefeeds are gone.

### What did you expect to see?

Once the last changefeed is deleted, TiCDC should remove its cluster-level service GC safepoint instead of leaving a residual safepoint entry behind.

After the last TiCDC node is shut down, there should be no stale TiCDC safepoint left in PD that can continue to constrain upstream GC.

### What did you see instead?

The cluster-level TiCDC service safepoint may remain in PD after all changefeeds are deleted.

There are two problematic outcomes:

1. If the last TiCDC node is shut down before the next GC reconcile tick, the stale safepoint can remain at an older stopped checkpoint.
2. Even if a final reconcile happens, TiCDC writes `currentTSO - 1` instead of deleting the service safepoint, so the safepoint entry can still remain in PD and then freeze until TTL expiration once the last TiCDC node is gone.

With the default `gc-ttl` this residual safepoint can survive for a long time and unexpectedly block upstream GC progress.

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
Not collected yet
```

Upstream TiKV version (execute `tikv-server --version`):

```console
Not collected yet
```

TiCDC version (execute `cdc version`):

```console
Not collected yet
```

### PR DESCRIPTION
**Pull request by hongyunyan**
### What problem does this PR solve?

Issue Number: close #4610

After the last changefeed is removed, TiCDC can still leave the cluster-level
service GC safepoint behind in PD. In the current code:

- the periodic reconcile path refreshes the safepoint to `currentTSO - 1`
  instead of deleting it when no changefeed remains;
- removing the last changefeed does not trigger an immediate cleanup, so the
  last coordinator can exit before the next GC tick and preserve the stale
  safepoint entry.

### What is changed and how it works?

- add `TryDeleteServiceGCSafepoint` to the GC manager and clear the cached GC
  state once the safepoint is deleted;
- change the classic-mode reconcile path to delete the cluster-level service GC
  safepoint when there is no remaining changefeed;
- make `RemoveChangefeed` issue a best-effort immediate delete after removing
  the last changefeed, which closes the shutdown window before the next GC tick;
- add regression tests for the no-changefeed reconcile path, the immediate
  delete after removing the last changefeed, and the GC manager cache reset.

#### Tests

- Unit test

Relevant verification:

- `make fmt`
- `go test ./pkg/txnutil/gc`
- `go test ./coordinator -run 'TestCreateChangefeedDoesNotUpdateGCSafepoint|TestUpdateGCSafepointCallsGCManagerUpdate|TestUpdateGCSafepointDeletesServiceSafepointWhenNoChangefeed|TestRemoveLastChangefeedDeletesServiceSafepointImmediately' -count=1`
- `go test ./coordinator/changefeed -run TestCalculateGCSafepoint -count=1`

Note:

- `go test ./coordinator -count=1` fails on this branch and on a clean
  `upstream/master` baseline because an existing coordinator test reuses fixed
  port `127.0.0.1:28300`.

##### Will it cause performance regression or break compatibility?

No. The change only affects the zero-changefeed cleanup path and reuses the
existing safepoint delete API.

##### Do you need to update user documentation, design documentation or monitoring documentation?

No.

### Release note

```release-note
Fix a bug where TiCDC could leave a stale cluster-level service GC safepoint in PD after the last changefeed was removed.
```

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved cleanup of cluster-level garbage-collection safepoints when the last changefeed is removed, reducing stale safepoints and avoiding blocking errors in classic kernel deployments.
  * Ensured GC state resets correctly after safepoint deletion to prevent lingering GC blocks.

* **Tests**
  * Added tests covering safepoint deletion, concurrent changefeed removal/creation, and related edge cases to increase reliability.

### CODE REVIEW

**Comment by hongyunyan** [comment]
Added a concurrent regression test for the remove-last-changefeed + create-new-changefeed handoff. It verifies that once the new changefeed is already visible, we do not delete the global GC safepoint, and the next GC reconcile updates it to the new checkpoint. I also replaced the open-ended operator wait in the existing test with a bounded require.Eventually check.

**Review by lidezhu — APPROVED** [review]

File: pkg/txnutil/gc/gc_manager.go [hunk 1]
```diff
@@ -110,6 +112,23 @@ func (m *gcManager) TryUpdateServiceGCSafepoint(
 	return nil
 }
 
+func (m *gcManager) TryDeleteServiceGCSafepoint(ctx context.Context) error {
```
[the remark this answers is not in the document]
Comment (hongyunyan) [reply pkg/txnutil/gc/gc_manager.go]: I did not take this one. TryUpdateServiceGCSafepoint(ctx, 0) is not equivalent to delete here: it still refreshes TiCDC's service safepoint with the normal TTL, while the delete path removes the service entry (classic mode ends up sending ttl=0 / max safepoint) and clears the cached "TiCDC is blocking GC" state immediately.

**Review by 3AceShowHand — APPROVED** [review]
